Automattic/jetpack
# Add Monorepo Installer Script

#### Changes proposed in this Pull Request:

Adds an installer script that will check and install all monorepo tooling. Installs up until "Running Jetpack Locally": https://github.com/Automattic/jetpack/blob/master/docs/quick-start.md


#### Jetpack product discussion


Internal: p1HpG7-dKb-p2 
#### Does this pull request change what data or activity we track or use?


No.
#### Testing instructions:





A bit tricky to test, but on a new environment, clone the Jetpack repo, CD into the monorepo root and run `./install-monorepo.sh`

Since it's best tested on machines that haven't used our tooling before, I was testing using an ubuntu installation via Multipass. To do so:

- install multipass: https://multipass.run/
- start a site: multipass launch --name test --disk 10G
- start a shell: multipass exec test -- bash
- use `sudo passwd ubuntu` to change the password. 
- clone the github repo
- cd into the cloned Jetpack folder.
- On your local machine, check out this branch and transfer the install-monorepo.sh file to the multipass setup. Standing on the monorepo root:
`multipass transfer install-monorepo.sh test:/jetpack/install-monorepo.sh`
- On your multipass environment, you may have to make the script executable: `chmod +x install-monorepo.sh`
- run `./install-monorepo.sh`
- Wait a long time for the installation to complete.
- Run `tools/check-development-environment.sh` and make sure everything looks good. Try some CLI commands, etc.
